include/struct_decls/message_data.h

```diff
@@ -0,0 +1,6 @@
+#ifndef POKEPLATINUM_STRUCT_MESSAGE_DATA_H
+#define POKEPLATINUM_STRUCT_MESSAGE_DATA_H
+
+typedef struct MessageData_t MessageData;
```

Comment:
Whatever we decided about _t, this is obviously wrong. `_t` stands for type, so it should be the suffix of any types, not struct names. i.e. `typedef struct MessageData MessageData_t;`. It's perfectly legal in C to use the same name in both a struct name and type name, or completely omit the struct name.